src/bus/ctl.rs

```diff
@@ -0,0 +1,486 @@
+use std::fmt::{self, Debug, Display, Formatter};
+use std::str::FromStr;
+
+use farcaster_core::{
+    blockchain::Blockchain,
+    role::TradeRole,
+    swap::btcxmr::{Offer, Parameters, PublicOffer},
+    swap::SwapId,
+};
+
+use amplify::{ToYamlString, Wrapper};
+use bitcoin::secp256k1::SecretKey;
+use bitcoin::Transaction;
+use internet2::addr::{InetSocketAddr, NodeAddr};
+use microservices::rpc;
+use strict_encoding::{NetworkDecode, NetworkEncode};
+
+use crate::bus::msg::Commit;
+use crate::bus::rpc::{AddressSecretKey, OfferStatus};
+use crate::bus::BusMsg;
+use crate::swapd::CheckpointSwapd;
+use crate::syncerd::SweepAddressAddendum;
+use crate::walletd::runtime::CheckpointWallet;
+use crate::{Error, ServiceId};
+
+#[derive(Clone, Debug, Display, From, NetworkEncode, NetworkDecode)]
+#[non_exhaustive]
+pub enum Ctl {
```

Comment:
The goal is to have `2.` with `Ctl` be the execution-relevant bus, your example with `Success`, etc. is half correct, this part needs a better refactor I left outside of this PR but:
- services, like `swapd`, needs to update the progress of a swap in `farcasterd`, so they use `Ctl` to instruct changes in fcd
- clients use the `Rpc` bus to read the progress

This is why we have such types in both buses

Reply:
got it, so having those messages on the `Ctl` bus too makes sense for upcoming ACL implementation since they mutate state in a service too, even if it's not relevant for protocol execution.

My current position is to leave as is and to further split up buses by services-only/including-external-agents only if our interfaces grow big enough to hurt code readability, or we want to impose very distinct ACLs for services-only/including-external-agents.